File: Modules/Default/MovieCast/MovieCastColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace SerenityProjem.Default.Columns;

[ColumnsScript("Default.MovieCast")]
[BasedOnRow(typeof(MovieCastRow), CheckNames = true)]
public class MovieCastColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MovieCastId { get; set; }
    public string MovieTitle { get; set; }
    public string PersonFullName { get; set; }
    [EditLink]
    public string Character { get; set; }
}

File: Modules/ServerTypes/Default/MovieCastColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MovieCastRow } from "./MovieCastRow";

export interface MovieCastColumns {
    MovieCastId: Column<MovieCastRow>;
    MovieTitle: Column<MovieCastRow>;
    PersonFullName: Column<MovieCastRow>;
    Character: Column<MovieCastRow>;
}

export class MovieCastColumns extends ColumnsBase<MovieCastRow> {
    static readonly columnsKey = 'Default.MovieCast';
    static readonly Fields = fieldsProxy<MovieCastColumns>();
}